task-forage.lic

```diff
@@ -0,0 +1,621 @@
+      false
+    when /I have a small item that needs/
+      echo 'Wrong task type' if @debug
+      decline_task
+      kill_time(2)
+      false
+    when /I am sorry, you must wait/
+      kill_time(1)
+      false
+    when /You are already on a task/
+      echo 'Already on a task. Canceling it' if @debug
+      cancel_task
+      kill_time(2)
+      false
+    end
+  end
+
+  def accept_task
+    echo 'accept_task' if @debug
+    case DRC.bput('accept task', 'Remember, I need', 'You have no offers to')
+    when /You have no offers to/
+      DRC.message('Task is missing.')
+      get_task
+    when /Remember, I need/
+      echo 'Task received' if @debug
+      DRCI.stow_hands
+    end
+  end
+
+  def decline_task
+    echo 'decline_task' if @debug
+    DRC.bput('decline task', 'I guess you do not wish to help', 'Please rephrase that command')
+  end
+
+  def cancel_task
+    echo 'cancel_task' if @debug
+    case DRC.bput("ask #{@task_giver} for task cancel", 'wish to end your current', 'do not currently have a',
+                  'frowns and shakes', 'To whom are you speaking?')
+    when /do not currently have a/, /frowns and shakes/
+      echo 'No task' if @debug
+      true
+    when /To whom are you speaking?/
+      find_giver
+      cancel_task
+    when /wish to end your current/
+      echo 'Cancel step 1' if @debug
+      case DRC.bput("ask #{@task_giver} for task cancel", 'I see that you no longer wish to assist',
+                    'do not currently have a')
+      when /do not currently have a/, /I see that you no longer wish to assist/
+        echo 'Cancel step 2' if @debug
+        true
+      end
+    end
+    kill_time(2) if determine_task_necessary
+  end
+
+  def use_boosts
+    echo 'use_boosts' if @debug
+    fput('info') # Check to see if a boost is already active
+    pause 0.25
+
+    lines = reget(50).reverse
+    lines.each do |line|
+      case line
+      when / +Task Master, expiring /
+        echo 'Already boosted' if @debug
+        @boosted = true
+        return true
+      when / +Encumbrance :/
+        echo 'Not boosted' if @debug
+        case DRC.bput('boost task', 'You\'ve activated the Task Master boost',
+                      'You don\'t have enough credits for the Task Master boost')
+        when /You've activated the Task Master boost/
+          echo 'Boost now activated' if @debug
+          @boosted = true
+          return true
+        when /You don't have enough credits for the Task Master boost/
+          echo 'Can\'t boost' if @debug
+          @boosted = false
+          return false
+        end
+      end
+    end
+  end
+
+  def kill_time(repetitions = 1)
+    echo "kill_time: #{repetitions}" if @debug
+    echo "use_boosts: #{@use_boosts}, and current status is #{@boosted}" if @debug
+    echo "never_wait: #{@never_wait}" if @debug
+    use_boosts if @use_boosts && @boosted.nil?
+
+    # No need to kill time if we're using a boost
+    return if !@boosted.nil? && @boosted
```

Comment:
I believe !boosted.nil? and @ boosted here would be the same condition. if it's nil, @ boosted would be falsey, if not, !@boosted.nil? would be true. so could just be return if @ boosted. Also, checking against nil wouldn't catch a false, right? so if @ boosted is false, the first condition would actually pass, whereas second condition would fail.